### ISSUE TITLE
Iris crashing with Physics Mod

### ISSUE THREAD

**Issue by RowanSkie** [IrisShaders/Iris#488]
### Iris and Minecraft Version

iris-mc1.17-1.1.0.jar for Minecraft 1.17 Fabric

### Are you running Sodium along with Iris?

No

### Operating System

Windows 10

### What is your GPU?

Nvidia GTX ZOTAC 1080ti

### Java Version

AdoptOpenJDK JRE Java 16.0.1

### Reproduction Steps

1. Run game.

### Crash Report file and latest.log

```shell
---- Minecraft Crash Report ----
// I'm sorry, Dave.

Time: 05/07/2021, 4:13 pm
Description: Initializing game

org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at Not Enough Crashes deobfuscated stack trace.(1.17+build.13)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:208)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:178)
	at org.spongepowered.asm.mixin.transformer.FabricMixinTransformerProxy.transformClassBytes(FabricMixinTransformerProxy.java:23)
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:162)
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:154)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at net.minecraft.client.MinecraftClient.<init>(MinecraftClient:575)
	at net.minecraft.client.main.Main.main(Main:179)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:234)
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:153)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:210)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:245)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)
Caused by: org.spongepowered.asm.mixin.throwables.MixinApplyError: Mixin [mixins.iris.yeetphysicsmod.json:MixinWorldRenderer] from phase [DEFAULT] in config [mixins.iris.yeetphysicsmod.json] FAILED during APPLY
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinError(MixinProcessor.java:642)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinApplyError(MixinProcessor.java:594)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:356)
	... 23 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.InvalidMixinException: Unexpecteded IllegalStateException whilst transforming the mixin class: [PostApply Phase -> mixins.iris.yeetphysicsmod.json:MixinWorldRenderer -> Plugin Post-Application]
	at Not Enough Crashes deobfuscated stack trace.(1.17+build.13)
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.postApply(MixinTargetContext.java:1204)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:339)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:345)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:569)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:351)
	... 23 more
Caused by: java.lang.IllegalStateException: Failed to redirect a method call in PhysicsMod rendering
	at net.coderbot.iris.compat.physicsmod.mixin.YeetPhysicsMod.redirect(YeetPhysicsMod.java:114)
	at net.coderbot.iris.compat.physicsmod.mixin.YeetPhysicsMod.postApply(YeetPhysicsMod.java:94)
	at org.spongepowered.asm.mixin.transformer.PluginHandle.postApply(PluginHandle.java:189)
	at org.spongepowered.asm.mixin.transformer.MixinInfo.postApply(MixinInfo.java:1355)
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.postApply(MixinTargetContext.java:1193)
	... 27 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:208)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:178)
	at org.spongepowered.asm.mixin.transformer.FabricMixinTransformerProxy.transformClassBytes(FabricMixinTransformerProxy.java:23)
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:162)
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:154)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	at net.minecraft.class_310.<init>(class_310.java:575)

-- Initialization --
Details:
Stacktrace:
	at net.minecraft.client.main.Main.main(Main.java:179)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:234)
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:153)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:210)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:245)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)

-- System Details --
Details:
	Minecraft Version: 1.17
	Minecraft Version ID: 1.17
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 16.0.1, AdoptOpenJDK
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode, sharing), AdoptOpenJDK
	Memory: 238499904 bytes (227 MiB) / 2067791872 bytes (1972 MiB) up to 7516192768 bytes (7168 MiB)
	CPUs: 8
	Processor Vendor: GenuineIntel
	Processor Name: Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz
	Identifier: Intel64 Family 6 Model 158 Stepping 9
	Microarchitecture: Kaby Lake
	Frequency (GHz): 4.20
	Number of physical packages: 1
	Number of physical CPUs: 4
	Number of logical CPUs: 8
	Graphics card #0 name: NVIDIA GeForce GTX 1080 Ti
	Graphics card #0 vendor: NVIDIA (0x10de)
	Graphics card #0 VRAM (MB): 4095.00
	Graphics card #0 deviceId: 0x1b06
	Graphics card #0 versionInfo: DriverVersion=30.0.14.7111
	Memory slot #0 capacity (MB): 8192.00
	Memory slot #0 clockSpeed (GHz): 2.40
	Memory slot #0 type: DDR4
	Memory slot #1 capacity (MB): 8192.00
	Memory slot #1 clockSpeed (GHz): 2.40
	Memory slot #1 type: DDR4
	Virtual memory max (MB): 29116.99
	Virtual memory used (MB): 16589.72
	Swap memory total (MB): 12800.00
	Swap memory used (MB): 1112.74
	JVM Flags: 5 total; -XX:+UnlockExperimentalVMOptions -XX:+UseShenandoahGC -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xms640m -Xmx7168m
	Fabric Mods: 
		appleskin: AppleSkin mc1.17-2.0.0
		architectury: Architectury 2.0.8
		autopath: Automatic Path 1.4.3
		betterbeds: Better Beds 1.2.0
		betterf3: BetterF3 1.1.5
		bettermineshafts: YUNG's Better Mineshafts 1.17-1.0.0
		betternether: Better Nether 5.1.0
		betterstrongholds: YUNG's Better Strongholds 1.17-1.0
		blue_endless_jankson: jankson 1.2.1
		cavebiomes: Cave Biomes 0.6.3
		cleardespawn: Clear Despawn 1.17-fabric0.34.9-1.1.1
		cloth-basic-math: Cloth Basic Math 0.5.1
		cloth-config2: Cloth Config v4 5.0.34
		com_moandjiezana_toml_toml4j: toml4j 0.7.2
		craftpresence: CraftPresence 1.8.2
		crowdin-translate: CrowdinTranslate 1.3+1.17
		cullleaves: Cull Leaves 2.2.0
		day_dream: Day Dream 1.0.2
		daylight_mobs_reborn: Daylight Mobs Reborn 1.2.2
		dehydration: Dehydration 1.1.6
		disable_custom_worlds_advice: Disable Custom Worlds Advice 1.3
		dynamicfps: Dynamic FPS 2.0.4
		dynmus: Dynamic Music 1.3.0
		fabric: Fabric API 0.36.0+1.17
		fabric-api-base: Fabric API Base 0.3.0+a02b44633d
		fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.2.0+84310a227d
		fabric-biome-api-v1: Fabric Biome API (v1) 3.1.11+c345aea83d
		fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.5+a02b44633d
		fabric-command-api-v1: Fabric Command API (v1) 1.1.1+a02b44633d
		fabric-commands-v0: Fabric Commands (v0) 0.2.2+92519afa3d
		fabric-containers-v0: Fabric Containers (v0) 0.1.12+a02b44633d
		fabric-content-registries-v0: Fabric Content Registries (v0) 0.2.2+a02b44633d
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.5+be9da3103d
		fabric-dimensions-v1: Fabric Dimensions API (v1) 2.0.10+a02b44633d
		fabric-entity-events-v1: Fabric Entity Events (v1) 1.1.0+a02b44633d
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.9+a722d8c03d
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.1+92519afa3d
		fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.6+a02b44633d
		fabric-item-api-v1: Fabric Item API (v1) 1.2.4+a02b44633d
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.2.10+b7ab61213d
		fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.4+a02b44633d
		fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.2+36b77c3e3d
		fabric-language-kotlin: Fabric Language Kotlin 1.6.2+kotlin.1.5.20
		fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 1.4.4+a02b44633d
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.0.4+a02b44633d
		fabric-mining-levels-v0: Fabric Mining Levels (v0) 0.1.3+92519afa3d
		fabric-models-v0: Fabric Models (v0) 0.3.0+a02b44633d
		fabric-networking-api-v1: Fabric Networking API (v1) 1.0.11+b7ab61213d
		fabric-networking-blockentity-v0: Fabric Networking Block Entity (v0) 0.2.11+a02b44633d
		fabric-networking-v0: Fabric Networking (v0) 0.3.2+92519afa3d
		fabric-object-builder-api-v1: Fabric Object Builder API (v1) 1.10.9+b7ab61213d
		fabric-object-builders-v0: Fabric Object Builders (v0) 0.7.3+a02b44633d
		fabric-particles-v1: Fabric Particles (v1) 0.2.4+a02b44633d
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.7.10+e2961fee3d
		fabric-renderer-api-v1: Fabric Renderer API (v1) 0.4.3+676f40fa3d
		fabric-renderer-indigo: Fabric Renderer - Indigo 0.4.8+a02b44633d
		fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 3.2.0+a02b44633d
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.1.5+a02b4463d5
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 0.1.13+a02b4463d5
		fabric-rendering-v0: Fabric Rendering (v0) 1.1.2+92519afa3d
		fabric-rendering-v1: Fabric Rendering (v1) 1.6.0+a02b44633d
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.4.7+b7ab6121d5
		fabric-screen-api-v1: Fabric Screen API (v1) 1.0.4+198a96213d
		fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.1.8+a02b44633d
		fabric-structure-api-v1: Fabric Structure API (v1) 1.1.11+e36969833d
		fabric-tag-extensions-v0: Fabric Tag Extensions (v0) 1.1.4+a02b44633d
		fabric-textures-v0: Fabric Textures (v0) 1.0.6+a02b44633d
		fabric-tool-attribute-api-v1: Fabric Tool Attribute API (v1) 1.2.12+b7ab61213d
		fabricloader: Fabric Loader 0.11.6
		fallingleaves: Falling Leaves 1.7.0+1.17
		fasttransferlib: Fast Transfer Lib 0.4.2
		gbfabrictools: GBfabrictools 1.3.4+1.17
		hydrogen: Hydrogen 0.3
		imm_ptl_core: Immersive Portals Core 0.7
		immersive_portals: Immersive Portals 0.7
		inventorysorter: Inventory Sorter 1.7.9-1.17
		iris: Iris 1.1.0+rev.f06a69d
		jankson: Jankson 3.0.1+j1.2.0
		java: OpenJDK 64-Bit Server VM 16
		kyrptconfig: Kytpt Config 1.1.10-1.17
		lambdabettergrass: LambdaBetterGrass 1.2.0+1.17
		lambdynlights: LambDynamicLights 2.0.1+1.17
		lazydfu: LazyDFU 0.1.2
		libgui: LibGui 4.0.0+1.17
		libninepatch: LibNinePatch 1.1.0
		lithium: Lithium 0.7.2
		mavm: More Axolotl Variants Mod 1.0.2b
		megane: megane 5.0.0
		megane-base: megane-base 5.0.0
		megane-fast-transfer-lib: megane-fast-transfer-lib 5.0.0+0.4.2
		megane-runtime: megane-runtime 5.0.0
		megane-team-reborn-energy: megane-team-reborn-energy 5.0.0+0.1.1
		megane-vanilla: megane-vanilla 5.0.0+1.17
		megane-wireless-networks: megane-wireless-networks 5.0.0+1.1.1
		midnightlib: MidnightLib 0.2.2
		minecraft: Minecraft 1.17
		modern_beta: Modern Beta Fabric 2.0.0-1.17
		modmenu: Mod Menu 2.0.2
		notenoughcrashes: Not Enough Crashes 3.3.1+1.17
		omega-config: OmegaConfig 1.0.8
		org_aperlambda_lambdajcommon: lambdajcommon 1.8.1
		org_jetbrains_kotlin_kotlin-reflect: kotlin-reflect 1.5.20
		org_jetbrains_kotlin_kotlin-stdlib: kotlin-stdlib 1.5.20
		org_jetbrains_kotlin_kotlin-stdlib-jdk7: kotlin-stdlib-jdk7 1.5.20
		org_jetbrains_kotlin_kotlin-stdlib-jdk8: kotlin-stdlib-jdk8 1.5.20
		org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm: kotlinx-coroutines-core-jvm 1.5.0
		org_jetbrains_kotlinx_kotlinx-coroutines-jdk8: kotlinx-coroutines-jdk8 1.5.0
		org_jetbrains_kotlinx_kotlinx-serialization-core-jvm: kotlinx-serialization-core-jvm 1.2.1
		org_jetbrains_kotlinx_kotlinx-serialization-json-jvm: kotlinx-serialization-json-jvm 1.2.1
		pehkui: Pehkui 2.2.1+1.14.4-1.17
		physicsmod: Physics Mod 1.0.0
		q_misc_util: A Miscellaneous Utility Library from qouteall 0.7
		repurposed_structures: Repurposed Structures 2.0.3+1.17.0
		roughlyenoughitems: Roughly Enough Items 6.0.251-alpha
		seamless_loading_screen: Seamless Loading Screen 1.3.5+1.17
		seasons: Fabric Seasons 1.2.2-BETA+1.17
		sodium: Sodium 0.3.0+IRIS1-rev.7491e3a
		spruceui: SpruceUI 3.2.0+1.17
		stoneholm: Stoneholm 1.3
		team_reborn_energy: Energy 0.1.1
		villagernames: Villager Names 2.0.0
		visualoverhaul: VisualOverhaul 3.3.0
		wirelessnetworks: Wireless Networks 1.1.1
		worldtime: WorldTime 1.17-fabric0.34.9-1.3.1
		wthit: wthit 3.6.1
		yungsapi: YUNG's API 1.17-Fabric-11
	Launched Version: MultiMC5
	Backend library: LWJGL version 3.2.2 build 10
	Backend API: NVIDIA GeForce GTX 1080 Ti/PCIe/SSE2 GL version 3.2.0 NVIDIA 471.11, NVIDIA Corporation
	Window size: <not initialized>
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: 
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	CPU: 8x Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz
	Suspected Mods: Fabric Loader (fabricloader), Iris (iris)
```

### Additional context

Latest.log with PhysicsMod and Iris: https://paste.ee/p/MynrM
Latest.log without Physics Mod and with Iris: https://paste.ee/p/ER2NU

I noticed a closed issue that states that this should be fixed already, but it doesn't seem to be like that.

**Comment by AlsoNiko** [comment]
This only happens with 1.17 physics mod, as a lot was changed

**Comment by AlsoNiko** [comment]
Should be fixed in recent commits

**Comment by coderbot16** [comment]
I don't think that this has been fixed yet.

**Comment by AlsoNiko** [comment]
This WAS fixed, and broke yet again with physics mod 1.17.1, 1.17 worked fine

**Comment by coderbot16** [comment]
Iris 1.1.2 works with recent versions of Physics Mod without crashes.
